apps/worker/src/app/workflow/services/metric-queue.service.ts

```diff
@@ -88,58 +88,63 @@ export class MetricQueueService extends QueueService<Record<string, never>> {
-            Logger.log(`MetricQueueService/${queueService.name}/failed`, JSON.stringify(failMetric));
-            Logger.log(`MetricQueueService/${queueService.name}/waiting`, waitCount);
-            Logger.log(`MetricQueueService/${queueService.name}/delayed`, delayedCount);
-            Logger.log(`MetricQueueService/${queueService.name}/active`, activeCount);
+        Logger.verbose('metric job started', LOG_CONTEXT);
+
+        try {
+          for (const queueService of this.token_list) {
+            const metrics = await queueService.bullMqService.getQueueMetrics();
+
+            const completeNumber = metrics.completed.count;
+            const failNumber = metrics.failed.count;
+
+            const successMetric: IMetric = {
+              count: metrics.completed.count,
+              total: completeNumber == 0 ? 0 : sum(metrics.completed.data),
+              min: completeNumber == 0 ? 0 : min(metrics.completed.data),
+              max: completeNumber == 0 ? 0 : max(metrics.completed.data),
+              sumOfSquares: completeNumber == 0 ? 0 : sumOfSquares(metrics.completed.data),
+            };
+
+            const failMetric: IMetric = {
+              count: metrics.failed.count,
+              total: failNumber == 0 ? 0 : sum(metrics.failed.data),
+              min: failNumber == 0 ? 0 : min(metrics.failed.data),
+              max: failNumber == 0 ? 0 : max(metrics.failed.data),
+              sumOfSquares: failNumber == 0 ? 0 : sumOfSquares(metrics.failed.data),
+            };
+
+            const waitCount = await queueService.bullMqService.queue.getWaitingCount();
+            const delayedCount = await queueService.bullMqService.queue.getDelayedCount();
+            const activeCount = await queueService.bullMqService.queue.getActiveCount();
+
+            if (process.env.NOVU_MANAGED_SERVICE === 'true') {
+              nr.recordMetric(`MetricQueueService/${queueService.name}/completed`, successMetric);
+              nr.recordMetric(`MetricQueueService/${queueService.name}/failed`, failMetric);
+              nr.recordMetric(`MetricQueueService/${queueService.name}/waiting`, waitCount);
+              nr.recordMetric(`MetricQueueService/${queueService.name}/delayed`, delayedCount);
+              nr.recordMetric(`MetricQueueService/${queueService.name}/active`, activeCount);
+            } else {
+              Logger.log(`MetricQueueService/${queueService.name}/completed`, JSON.stringify(successMetric));
+              Logger.log(`MetricQueueService/${queueService.name}/failed`, JSON.stringify(failMetric));
+              Logger.log(`MetricQueueService/${queueService.name}/waiting`, waitCount);
+              Logger.log(`MetricQueueService/${queueService.name}/delayed`, delayedCount);
+              Logger.log(`MetricQueueService/${queueService.name}/active`, activeCount);
+            }
           }
+
+          return resolve();
+        } catch (error) {
+          return reject(error);
```

Comment:
Catching to avoid unexpected unhandled exceptions. This could be logged only.

Reply:
I expect this rejection to be caught by the `failed` event (line 35) and to be logged through `jobHasFailed` function. But not 100% sure.

Reply:
@p-fernandez @djabarovgeorge yes, this is what will happen after `reject`